### PR TITLE
Enable conditional checking of values in the Rust codebase

### PR DESCRIPTION
**Pull request by Urgau**
This pull-request enable conditional checking of (well known) values in the Rust codebase.

Well known values were added in https://github.com/rust-lang/rust/pull/94362. All the `target_*` values are taken from all the built-in targets which is why some extra values were needed do be added as they are not (yet ?) defined in any built-in targets.

r? @Mark-Simulacrum

### CODE REVIEW

File: src/bootstrap/lib.rs [hunk 1]
```diff
@@ -199,6 +199,15 @@ const EXTRA_CHECK_CFGS: &[(Option<Mode>, &'static str, Option<&[&'static str]>)]
     (Some(Mode::Std), "no_global_oom_handling", None),
     (Some(Mode::Std), "freebsd12", None),
     (Some(Mode::Std), "backtrace_in_libstd", None),
+    /* Extra values not defined in the built-in targets yet, but used in std */
+    (Some(Mode::Std), "target_env", Some(&["libnx"])),
+    (Some(Mode::Std), "target_os", Some(&["tvos", "watchos"])),
```
Comment (Mark-Simulacrum) [inline src/bootstrap/lib.rs:204]: compiler/rustc_target/src/spec/aarch64_apple_tvos.rs seems to define tvos in a built-in target, and watchos appears to not be used anywhere in std?

**Comment by Urgau** [comment]
cc @petrochenkov: the above failure is due to the `unexpected_cfg` lint fire-ring on `#[cfg(target_pointer_width = "8")]` as "8" is not an expected value. I could add "8" to the list of extra values for std but I wonder if it wouldn't be better to set the list of `target_pointer_width` to 8, 16, 32, 64 and 128 (the current one are: 16, 32, 64) in the compiler instead of relying on the built-in targets for it ?

**Comment by petrochenkov** [comment]
`target_pointer_width = "8"` looks like a mistake, I'm not aware of any out-of-tree targets using it, and there's a bunch of places in the compiler that produce asserts on any pointer width except for `16 | 32 | 64`.

**Comment by Urgau** [comment]
Just in case you haven't seen that it was ready for another review.

@rustbot ready

**Pull request merged**